Sources/SpeziHealthKit/Sample Types/HealthKitCharacteristic.swift

```diff
@@ -107,21 +110,43 @@ extension HealthKitCharacteristicProtocol where Self == HealthKitCharacteristic<
 
 extension HealthKitCharacteristicProtocol where Self == HealthKitCharacteristic<Date> {
     /// The characteristic representing the user's date of birth.
+    ///
+    /// The `Date` values returned here are always implicitly adjusted to match midnight in the current time zone,
+    /// regardless of whether the resulting date would still fall into the same day as the original birthday.
+    /// This behaviour matches the instinctive way most people treat birthdays: if you were born in the morning on may 17th in asia,
+    /// your birthday is still may 17th when you move to the united states.
+    ///
+    /// - Note: Prefer using ``dateOfBirthComponents`` if you need the user's date of birth in a non-UI-related context.
+    ///     The `DateComponents` will have their `calendar` and `timeZone` properties set based on the specific time zone in which the user entered their date of birth.
     public static var dateOfBirth: HealthKitCharacteristic<Date> {
         Self(.dateOfBirth, displayTitle: "Date of Birth") { healthStore in
             let components = try healthStore.dateOfBirthComponents()
-            if let date = Calendar.current.date(from: components) {
-                // Question: Do we need to take time zones into account here?
-                // What if the user entered their DoB in a different time zone than the one they're currently in?
+            if let date = Calendar.current.date(from: .init(year: components.year, month: components.month, day: components.day)) {
```

Comment:
ideally we'd use the `Calendar` and/or `TimeZone` from the querying View's environment here, instead of just doing `Calendar.current`, but this is probably good enough for the time being.

we could actually achieve this fairly easily, since the view's environment is available to the property wrapper which ends up calling this function